src/webapi/window_or_worker.rs

```diff
@@ -29,4 +31,32 @@ pub trait IWindowOrWorker: ReferenceType {
             }, $3 );\
         ", self.as_ref().as_raw(), funcall_adapter::< F > as extern fn( *mut F ), callback, timeout );
     }
+
+    /// Sets a timer which executes a function once after the timer expires and can be cleared
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout)
+    // https://html.spec.whatwg.org/#windoworworkerglobalscope-mixin:dom-settimeout
+    fn set_clearable_timeout< F: FnOnce() + 'static >( &self, callback: F, timeout: u32 ) -> TimeoutHandle {
+        //let callback = Box::into_raw( Box::new( callback ) );
+        let callback_reference: Reference = js! ( return @{callback}; ).try_into().unwrap();
```

Comment:
You need to use the wrapper from `webcore/once.rs` to pass an `FnOnce` into `js!`.